core/module.cc

```diff
@@ -84,17 +85,21 @@ Module *ModuleBuilder::CreateModule(const std::string &name,
   m->set_name(name);
   m->set_module_builder(this);
   m->set_pipeline(pipeline);
-  module_graph_.emplace(std::piecewise_construct, std::forward_as_tuple(name),
-                        std::forward_as_tuple(m));
-  if (m->IsTask()) {
-    task_graph_.emplace(std::piecewise_construct, std::forward_as_tuple(name),
-                        std::forward_as_tuple(m));
-  }
   return m;
 }
 
 bool ModuleBuilder::AddModule(Module *m) {
-  return all_modules_.insert({m->name(), m}).second;
+  if (m->is_task_) {
+    if (!tasks_.insert(m->name()).second) {
+      return false;
+    }
+  }
+  return all_modules_.insert({m->name(), m}).second &&
```

Comment:
I would break this up so it reads like a sentence.  E.g.:
`
bool module_added = all_modules_.insert({m->name(), m}).second;
bool graph_updated = ...
return module_added && graph_updated;
`